extra_foam/utils.py

```diff
@@ -42,6 +42,31 @@ def timed_f(*args, **kwargs):
     return wrap
 
 
+
+class BlockTimer:
+    """A context manager for measuring the execution time of a code block
+
+    For example::
+
+        >>> with BlockTimer("foo"):
+        ...     time.sleep(1)
+        ...
+        Execution of foo: 1.001s
+    """
+    def __init__(self, label="block", enabled=True):
+        self._label = label
+        self._enabled = enabled
+
+    def __enter__(self):
+        self.start = time.time()
+        return self
+
+    def __exit__(self, *_):
+        duration = time.time() - self.start
+        if self._enabled:
+            print(f"Execution of {self._label}: {duration:.3f}s")
```

Comment:
I am kinda concerned with the resolution. While this suffice for big and slow blocks, one might find it useful to measure quick blocks. Should you like to extend it, perhaps `time.perf_counter` might be more fitting.

Reply:
Good point, fixed in aa779423.